Lib/test/test_capi/test_codecs.py

```diff
@@ -516,5 +523,290 @@ def test_asrawunicodeescapestring(self):
         # CRASHES asrawunicodeescapestring(NULL)
 
 
+class CAPICodecs(unittest.TestCase):
+
+    def setUp(self):
+        # Encoding names are normalized internally by converting them
+        # to lowercase and their hyphens are replaced by underscores.
+        self.encoding_name = f'codec_reversed_{id(self)}'
```

Comment:
Please use a simple name, otherwise tests for reference leaks become crazy.

```suggestion
        self.encoding_name = 'test_codec_reversed'
```